### ISSUE TITLE
Please add email support or help me resolve username issue

### ISSUE THREAD

**Issue by davidkoplik** [samczsun/Skype4J#128]
I created an account just for my Skype Bot, but I see it only supports email as a username. Is there a way to login into Skype with your Skype Bot with email? Because I think there is no way I can get a normal username anymore other than email.

**Comment by ducc** [comment]
Your username is probably in this format: `live:username`
Login with that full string & it should work. If you can't find your username ask a friend to get it for you.

**Comment by solo-yolo** [comment]
i'm experiencing similar issue.
i've recently created new account for bot and it got username in format
`live:email-without-domain`
and while i am able to login to https://web.skype.com with those credentials, getting annoying error  `{"status":{"code":40120,"text":"Authentication failed. Bad username or password."}}`
when doing same from code.
anyone knows how to solve this?

**Comment by nanda1505** [comment]
I am experiencing the same issue.  I have created a new Skype Account for the BOT using email as the login ID.   I am able to login to web.skype.com using the email as the username....  But the moment I use try to use the same credentials in my BOT code,  its giving me the following error. 
{"status":{"code":40120,"text":"Authentication failed. Bad username or password."}}.

Also,  when I am trying to access the username,  I am not able to get the same.

Waiting for the response/suggestion/solution...

**Comment by ducc** [comment]
Not a solution, but if you create a new skype account on IOS it doesn't have the `live:` prefix.

**Comment by nanda1505** [comment]
you can create an account using Email Id or using your Mobile Number....  I had tried creating an account using Email Id,  and I am not able to connect to it.   Should I try now for Mobile Number....

**Comment by itsGhost** [comment]
https://github.com/itsGhost/Skype-testing/blob/master/XTokenFromMicrosoftAccountLogin.java
> Extend FullClient
> Override login()
> Call  getXtoken(email, pass) 
> new MSFTClient("test@mail.com", "password", new HashSet<String>(), null, new ArrayList<ErrorHandler>());

**Comment by GalHorowitz** [comment]
@itsGhost Could you explain? I didn't udnerstand what I should do...

**Comment by itsGhost** [comment]
Learn the basics of oop before crying for java assistance on a library thread because you too lazy to either learn and/or google it

**Comment by GalHorowitz** [comment]
@itsGhost  I do know what I'm doing. The structure is not what I'm looking for. getXtoken always return null for some bizzare reason.

**Comment by itsGhost** [comment]
I literally used a similar implementation two weeks ago. Since you claim to know what you're doing there's no need of supporting you. Sorry "mr know what I'm doing," you clearly don't, if you need someone to tell you to debug/bp/print and use common sense.

**Comment by Manevolent** [comment]
There really is no need to be an ass to someone else who needs help. Be respectful.

**Comment by itsGhost** [comment]
Be respectful? How about dont be a fucking idiot and lie about your skill sets? I explained what you had to do like you was five, and then if one cant follow, you're just an idiot

**Comment by Manevolent** [comment]
You honestly come across as a five year-old with how aggressive you are around here. In other news I might put together a pull request for this if this hasn't been fixed yet

**Comment by itsGhost** [comment]
I don't give a fuck with what you plan on doing or how much shit you are full of. Just do it instead of baiting or otherwise teasing. I've already linked the necessary code for anyone who wants to implement a work around. 
Ps: calling people 5 year olds is pretty much void when you have a childrens cartoon profile

**Comment by itsGhost** [comment]
Ps 2: there's pretty much zero links between aggressiveness and age

**Comment by Manevolent** [comment]
You need to calm down, just get the issue resolved and not dirty it with spam. Thanks for the fix regardless, saves me time.

**Comment by itsGhost** [comment]
I'd love to get the issue resolved, but this last year sam has been hell bent on ignoring this projects issues and refusing stuff that requires him to actually do something. Now I get notifications every week because people are having issues with code someone pretty much c/p'd from a dev chat. If I could be bothered to restore my old skype db, Id have chat logs of sam (the self-proclaimed reverse engineer that copies from an open wiki) saying he'll implement this soon several months ago (if not last year)

**Comment by samczsun** [comment]
Have you considered that the fact I'm inactive across all projects means that I might be busy with other, more important, stuff? Open source is nice if there's time but when I have other commitments I do those first.

I'm aware of the state of this project (i.e. broken) but until I get some time freed up there's not much I can do about it. If you want to flame and/or make claims with no factual basis (i.e. my copying from an open wiki) then I'll be happy to show you the door.

Please don't consider arguing with my reply. I'll work on this when I have the time. That's the way it's always been and it's the way it's going to stay. You can choose to disagree but try to keep the disagreement civil or I'll be forced to start deleting the more vulgar comments.

**Comment by itsGhost** [comment]
TL;DR - Stop lying and see the proof section.

A) It's hard to believe that when you've been using the same generic excuse since last August. You've clearly had free time as you started working on a Java decompiler/deobfuscator whilst you've been "busy." Looking at the Java OPcode table, there's a lot of stuff to implement, and you was quite optimistic about completing that project. All that indicates to me and many others is that you have been lying to our faces regarding your schedule. 

B) Don't fucking lie to my face. You literally was sending me links from some skype web wiki a few months back in-regards to online presences because you haven't been able to implement them

C) Censor everything! It's always one way to prove your superiority and argument. 

PS: Where have you been on every other issue or do you only popout when someone starts to call you out?

Proof of you just leaching of someones wiki:  https://reece.sx/3PpfcOShjiRALQi

**Comment by ducc** [comment]
xd this got heated xd

**Comment by GalHorowitz** [comment]
The thing that makes me laugh, is that itsGhost has no right to be angry, sam is doing this for free, open source, in his own time. He doesnt owe any of us anything. If he doens't want to work on this, he can chose so.

**Comment by itsGhost** [comment]
a) it doesn't concern you therefore you have no right to even comment 
b) im not angry
c) do i have to check my privilege before i talk now, sjw nut job?
e) getting lied to does justify anger

**Comment by GalHorowitz** [comment]
a) When you call me an idiot it does
b) Well it seems like it from the last 5 raging comments
c) Seems like all you know is how to curse, so keep on going
d) Even if he promised he would work on it, he doesnt owe you anything. He can right here right now delete this repository and you can go search.

**Comment by itsGhost** [comment]
> a) When you call me a an idiot it does
What? You are. Don't ask stupid questions then.

> b) Well it seems like it from the last 5 raging comments
You not liking what I'm saying doesn't equate to anger 

> c) Seems like all you know is how to curse, so keep on going
Neither my original message or my updated message contained more than a swear word. If you're just concerned about a certain word being used, and you somehow think that makes someones point null & void, once again, you're showing signs of an idiot.

> d) Even if he promised he would work on it, he doesnt owe you anything. He can right here right now delete this repository and you can go searc
No one fucking said that, you deaf twat. You complain that I swear then you come out with the most outrageous, ideological nonsense. Again, "It being free nor opensource (or the opposite) doesn't ensure there will be updates." If you bothered to even read any of my posts, you'd know my point is go implement it yourself and sam is a lair and a fraud. The last thing he needs are gay white nights coming to his rescue, as he is a grown man, he can deal with this himself. Go book an appointment with your doctor as your eyes clearly aren't working. Or perhaps, better yet, go back to playing minecraft with poorly developed plugins.

**Comment by itsGhost** [comment]
> xd this got heated xd

kek

**Comment by GalHorowitz** [comment]
Whatever you want kiddo. You keep on arguing but at the end of the road you are raging and I'm completely fine.

**Comment by itsGhost** [comment]
> Whatever you want kiddo. You keep on arguing but at the end of the road you are raging and I'm completely fine.

Do you have any better strawmans or are you running out? You have a minecraft profile picture, have shit code plastered everywhere, fail to read posts, edit messages after I mention censorship is silly, then cry I'm a kid. Alrighty. Just a little reminder, no matter how much you scream kid, it will not help your point in anyway. You clearly care enough to jump into a conversation you do not belong in.

**Comment by GalHorowitz** [comment]
Whatever makes you feel better.

**Comment by samczsun** [comment]
1) I'm neither obligated or required to divulge my entire schedule to you, nor am I ever going to. If I don't want to tell you my schedule, I'm going to tell you that. If you still won't stop annoying me, I will lie to you so you shut up. If you think the world revolves around you and your need for stuff to get done, you need to wake up. I start projects if I want, and I'll pause projects if I want too. It's my project, and therefore my rules apply.

2) Explain how I implemented stuff like the banlist, topic, and active status - things which weren't in the web client at the time? Actually, don't. I don't want to hear it.

3) I will gladly censor everything, starting now. This issue thread could have been a clean issue thread like any other, until you decided to turn around and attack everyone else here. I realise that there are no 'rules' which have been set in place but I expect that common sense should prevail, which it clearly didn't.

As a result, you are no longer welcome in this project, or any other projects or mine. Please refrain from posting any replies, especially of the snarky type.

When this reply has been posted, I will be going through all of your comments in this thread, and I will be editing out the vulgarity. If you post any more replies, they will be deleted. That is all.

**Comment by Manevolent** [comment]
Just a heads up, the fix provided does indeed work, but it needs XML sanitation since it's just a straightforward String.format, which can be found in apache commons-lang3 if anyone is curious. 

You'll also need to use the skypename property returned from the json in https://api.skype.com/rps/skypetoken, as I was having trouble with some requests without it set correctly in the client (I was testing with a migrated skype).
